pandas/tests/series/test_constructors.py

```diff
@@ -142,6 +142,22 @@ def test_constructor_list_like(self):
             result = Series(obj, index=[0, 1, 2])
             assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize('input_vals, dtype, expected', [
+        ([1, 2, 3], 'str', ['1', '2', '3']),
```

Comment:
can you add other dtypes here: strings, datetimes, datetime w/tz, period, timedelta, categorical (with int), categorical (with float), categorical (with str), interval. throw in a nan or two on the floats.

(you can make it an array of len 2 to type a bit less if it helps). could also parametrize this I think (IOW an astype of str of a list should be quiv of str on each element

Reply:
I added some more dtypes to the tests. I'm not sure how to add a test for categorical, though, because ``Series(..,dtype=str)`` throws a ValueError for a Categorical but ``.astype(str)`` doesn't:
```
In [1]: pd.Series(pd.Categorical([1.0, 2.0, np.nan])).astype(str)
Out[1]: 
0    1.0
1    2.0
2    nan
dtype: object

In [2]: pd.Series(pd.Categorical([1.0, 2.0, np.nan]), dtype=str)
Out[2]:
ValueError: cannot specify a dtype with a Categorical unless dtype='category'
```